scripts/install-dotnet.sh

```diff
@@ -2,7 +2,7 @@
 # print command before they are executed without expanding variables
 set -v
 
-# install the latest patch version of 3.1
-curl -sSL https://dot.net/v1/dotnet-install.sh | bash /dev/stdin -c 3.1
+# install the latest patch version of 6.0
+curl -sSL https://dot.net/v1/dotnet-install.sh | bash /dev/stdin -c 6.0
```

Comment:
Why did the version need to be updated?

Reply:
.NET 3.0 has reached end of life and is no longer supported. Changing to .NET 6.0 which is a supported release.

Reply:
`.NET Core 3.1` is indeed old, but technically still maintained.

If you've tested with `6.0` and nothing broke, then that's fine by me as well!

Reply:
All tests pass with .NET 6. This change is not essential but end of life for .NET 3.1 is in 3 months.